### ISSUE TITLE
Optimize transit edge formater function format_transit_net_edge() for speed

### ISSUE THREAD

**Issue by sablanchard** [UDST/urbanaccess#3]
The format_transit_net_edge() function currently operates with a loop that is not optimized for computational speed. The loop can be found here: https://github.com/UDST/urbanaccess/blob/master/urbanaccess/gtfs/network.py#L382-L418 and can potentially be vectorized and improved upon for speed.

**Comment by Eh2406** [comment]
Do we have a reproducible code to exercise this slowdown?

**Comment by sablanchard** [comment]
Hi Jacob,
Thank you for your interest. @pksohn and I are planning to try some techniques next week to speed it up. Let us try on our end first and we can keep you posted on our progress in this issue. The information that can be used to get started can be found below:

Here are the interpolated stop time data that is expected as input for the `stop_times_df` parameter here: https://dl.dropboxusercontent.com/u/515309/ac_transit_stop_times_int_subset.h5

Example code:
```
import pandas as pd
from urbanaccess.gtfs.network import format_transit_net_edge

stop_times_df = pd.read_hdf('data/ac_transit_stop_times_int_subset.h5', key='stop_times_int')

edge_table = format_transit_net_edge(stop_times_df=stop_times_df[['unique_trip_id',
                                                                  'stop_id',
                                                                  'unique_stop_id',
                                                                  'timediff',
                                                                  'stop_sequence',
                                                                  'unique_agency_id',
                                                                  'trip_id']], verbose=False)

```

You can step through the code and process one trip at a time to get a feel for what it is doing.

**Comment by Eh2406** [comment]
Ok, I will mostly leave working on this till you give me the word. :-)

Just some comments from a quick read. Note: it is a quick read so I may be wrong about anything.
1. in the `else` branch if one substitutes `i = 0`, one gets the exact same thing as the `i = 0` branch. Ether the branch is unnecessary or the logic is missing something. 
2. I think the `unique_stop_id_row` loop can be vectorized as 
```python
        i_rang = edge_df.index
        edge_df['node_id_from'] = tmp_trip_df['unique_stop_id'].iloc[i_rang].values
        edge_df['node_id_to'] = tmp_trip_df['unique_stop_id'].iloc[i_rang + 1].values
        edge_df['weight'] = tmp_trip_df['timediff'].iloc[i_rang + 1].values
        edge_df['unique_agency_id'] = tmp_trip_df['unique_agency_id'].iloc[i_rang + 1].values
```
3. `merged_edge_df['sequence'] = ...` and `merged_edge_df['id'] = ...` only need to be done once at the end, instead of once per `for trip` loop.

With all of those changes I can run the sample file in 15 seconds. As I said I have not read the code carefully so please double check that my changes are equivalent.

**Comment by pksohn** [comment]
@Eh2406 Nice work! Since you've already made so much progress, do you want to submit a PR to the `edge_formatter` branch? Then we can review, and we can send comments about any necessary changes for you to revise (if you want), or can make the changes directly ourselves.

**Comment by Eh2406** [comment]
working on a pr. The slow part is making absolutely sure the output is the same.

**Comment by sablanchard** [comment]
Fixed with #4
